packages/ui/src/theme/utils.ts

```diff
@@ -34,20 +34,20 @@ function referenceValue(value: string) {
   return value;
 }
 
-export function cssValue(token: DesignToken<{ value: unknown }>) {
+export function cssValue(token: BaseDesignToken): string | number {
   const { value } = token;
   if (isString(value)) {
     return referenceValue(value);
   }
 
-  if (isObject(value)) {
-    if ('offsetX' in value) {
-      return SHADOW_PROPERTIES.map((property) =>
+  if (isShadowToken(value) && isObject(value)) {
+    return SHADOW_PROPERTIES.map((property) => {
+      return referenceValue(
         // lookup property against `token` first for custom non-nested value, then lookup
         // property against `value` for design token value, default to empty string
-        referenceValue(token[property] ?? value[property] ?? '')
-      ).join(' ');
-    }
+        isShadowToken(token) ? token[property] : value[property] ?? ''
```

Comment:
Silly question incoming: Does the nullish coalescing get applied after the ternary resolves or is it only applied to the right condition?

Reply:
Ok I think I made it better by removing the nullish coalescing and making `referenceValue` return empty string if its input is nullish.